### PR TITLE
Wrap the GroupRecord instead of making a full copy before insertion

### PR DESCRIPTION
**Pull request by remcowesterhoud**
## Description

Wrapping is more performant than copying. In this place we don't need a full copy as we immediately insert it in our ColumnFamily.

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

relates to https://github.com/camunda/camunda/pull/25635#pullrequestreview-2481712329

### CODE REVIEW

**Comment by remcowesterhoud** [comment]
@korthout any chance I could get a review on this very small PR?

File: zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/group/PersistedGroup.java [hunk 1]
```diff
@@ -23,6 +24,11 @@ public PersistedGroup() {
     declareProperty(groupKeyProp).declareProperty(nameProp);
   }
 
+  public void wrap(final GroupRecord group) {
+    groupKeyProp.setValue(group.getGroupKey());
+    nameProp.setValue(group.getName());
```
Comment (korthout) [inline zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/group/PersistedGroup.java:29]: 🔧 The `getName` method makes a string from the buffer, you can use `getNameBuffer()` to directly set the name buffer

**Comment by remcowesterhoud** [comment]
Since this PR the `tenantIdsProp` got added to the `PersistedGroup`. Since a `GroupRecord` doesn't contain a list of tenant ids it makess sense to reset this array here. The wrap method is only called on creation at this moment. The list of tenant ids should always be empty at that point in time.

**Comment by korthout** [comment]
>he wrap method is only called on creation at this moment. The list of tenant ids should always be empty at that point in time.

@remcowesterhoud But shouldn't the caller of `wrap` just trust that works regardless of when it's called? Why not just set the tenants as well?

**Comment by remcowesterhoud** [comment]
> >he wrap method is only called on creation at this moment. The list of tenant ids should always be empty at that point in time.
> 
> 
> 
> @remcowesterhoud But shouldn't the caller of `wrap` just trust that works regardless of when it's called? Why not just set the tenants as well?

Because there is no list of tenants in the group record we are wrapping. So there's nothing to set.

**Review by korthout — APPROVED** [review]
>Because there is no list of tenants in the group record we are wrapping. So there's nothing to set.

@remcowesterhoud Oh wow! I see now that these are completely different entities with different properties. A `copyFrom` only worked because they share some property names.

LGTM 👍

**Pull request merged**